### ISSUE TITLE
New Rule: prefer-console-formatting

### ISSUE THREAD

**Issue by ericbn** [eslint-community/eslint-plugin-n#301]
### Rule details

Prefer using a string with formatting elements (or just using separate arguments) instead of concatenating strings or using template literals for the first argument of console.log, console.warn, console.error, etc.

### What type of rule is this?

Suggests an alternate way of doing something

### Example code

```js
console.log('count: ' + count);
console.log(`count: ${count}`);
console.error("error #" + code);
console.error(`error #${code}`);
```

### Participation

- [X] I am willing to submit a pull request to implement this rule.

**Comment by aladdin-add** [comment]
This rule seems quite specific and might reflect personal preferences more than general best practices. It is suggested that you implement this rule yourself if it aligns closely with your project's style guide or team preferences.

@eslint-community/eslint-plugin-node thoughts?

**Comment by scagood** [comment]
I am hesitant to add something like this as, I think:
1. this would get very complex rather quickly
2. it's better to let people pass whatever they want to `console` (do we also error on variables going in too, just because they could be a constructed string)?

---

The first example could be converted like this, but count looks like it should be `%i` or `%d`, not `%s`:
```ts
console.log('count: %s', count);
// or
console.log('count:', count);
```

---

The second example could be converted like this:
```ts
console.error('error #%s', code);
```

---

Another example could be:
```ts
console.error(`Good morning ${name}\n Welcome back`);
console.error('Good morning %s\n Welcome back', name);
```

**Comment by ericbn** [comment]
Thank you for the comments.

I see there's no added benefit between
```
console.log('count: ' + count);
console.log(`count: ${count}`);
console.log('count: %d', count);
console.log('count:', count);
```
as in all cases the parameters will be evaluated and the message will be logged.

Closing this.
